unit_tests/sources/declarative/decoders/test_composite_decoder.py

```diff
@@ -117,3 +118,30 @@ def test_composite_raw_decoder_jsonline_parser(requests_mock, encoding: str):
     for _ in composite_raw_decoder.decode(response):
         counter += 1
     assert counter == 3
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        ({"data-type": "string"}),
+        ([{"id": 1}, {"id": 2}]),
+        ({"id": 170_141_183_460_469_231_731_687_303_715_884_105_727}),
+    ],
+    ids=[
+        "test_with_buffered_io_base_data_containing_string",
+        "test_with_buffered_io_base_data_containing_list",
+        "test_with_buffered_io_base_data_containing_int128",
+    ],
+)
+def test_json_parser_with_valid_data(data):
+    encodings = ["utf-8", "utf", "iso-8859-1"]
+
+    for encoding in encodings:
+        raw_data = json.dumps(data).encode(encoding)
+        for i, actual in enumerate(
+            JsonParser(encoding=encoding).parse(BufferedReader(BytesIO(raw_data)))
```

Comment:
nit: could you please wrap it in requests_mock and use CompositeRawDecoder.decode interface?

will approve 👍  after this